### ISSUE TITLE
Black fails with "AST error message: invalid syntax" if using the walrus operator in parenthesis

### ISSUE THREAD

**Issue by brmc** [psf/black#1262]
**Describe the bug** Black cannot format code that uses the walrus operator, `:=` , in parentheses

**To Reproduce** Steps to reproduce the behavior:

1. Take this file:

```python

if (blah := 4):
    print(666)
```
2. Run _Black_ (v19.10b0) on it with default arguments
3. It should produce the following error: `error: cannot format example.py: cannot use --safe with this file; failed to parse source file.  AST error message: invalid syntax (<unknown>, line 1)
`

**Expected behavior** It should format the file and not throw an error

**Environment (please complete the following information):**

- Version: 19.10b0
- OS and Python version: Linux Mint 19.3, python 3.8.0

**Does this bug also happen on master?** No

**Comment by zsol** [comment]
I can't repro this either locally or on https://black.now.sh so there's something likely with your environment. I would check what version of python Black is running with. You'd get this message if you're running it with something <3.8

**Comment by cr-mocz** [comment]
I can't reproduce this either.  Tested with black 19.10b0 on Mint 19 and Python 3.8.1

✗ cat 1262.py 
if (blah := 4):
    print(666)

✗ black .
reformatted /home/cr/cloud-files/Repos/lessons-learned/1262.py
All done! ✨ 🍰 ✨
1 file reformatted, 17 files left unchanged.

✗ cat 1262.py 
if (blah := 4) :
    print(666)

**Comment by mtolkacz-lingaro** [comment]
No need to reopen an issue, but I will share my case here.

I had similar issue, but the indicated line of the code by black was not a real issue. 
I used VS Code. It was restarted a few times (after reinstalling lib, upgrading etc.).
I checked the script on the Black Playground website, which told me where was **the real issue.** 

Black Playground had newer version of Black (20.8 vs 22.3) so it might be helpful to just upgrade the lib in the Pip, BUT in my case, what is really strange, the same version locally (black-22.3.0) still showed an old error.
I formatted the file online, then pasted it to the IDE, and problem was gone.
